flake.nix

```diff
@@ -1,27 +1,73 @@
 {
   description = "A collection of templates for development projects";
 
-  outputs = { self }: {
-    templates = {
-      latex = {
-        path = ./latex;
-        description = "A LaTeX project";
-      };
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    devshell.url = "github:numtide/devshell";
+    treefmt.url = "github:numtide/treefmt";
```

Comment:
Should we consider adding treefmt to each template instead? We can then place it in the checks script for each template.

Reply:
Also note that `treefmt` does not provide outputs for `aarch64` systems, so maybe hold off on including it for now.